vscode/src/debugger/session.ts

```diff
@@ -144,57 +190,392 @@ export class QscDebugSession extends LoggingDebugSession {
     });
     await Promise.race([configurationDone, delay(ConfigurationDelayMS)]);
 
-    const programText = await this._fileAccessor.readFileAsString(args.program);
-
-    const diagnostics = await this._compiler.checkCode(programText);
-    if (diagnostics.length > 0) {
-      log.trace("compilation failed. sending error response");
-      this.sendErrorResponse(response, {
-        id: -1,
-        format: ErrorProgramHasErrors,
-        showUser: true,
-      });
-      this.sendResponse(response);
-      return;
-    }
-
     if (args.noDebug) {
       log.trace(`Running without debugging`);
-      this.sendResponse(response);
-      await delay(LaunchDelayMS);
-      this.runWithoutDebugging(args);
+      await this.runWithoutDebugging(args);
     } else {
       log.trace(`Running with debugging`);
-      this.sendResponse(response);
-      // This is where we would start the debugger
-      await delay(LaunchDelayMS);
-      this.runWithoutDebugging(args);
+      await this.runWithDebugging(args);
     }
+    log.trace(`sending launchRequest response`);
+    this.sendResponse(response);
   }
-
-  private async runWithoutDebugging(
-    args: ILaunchRequestArguments
+  private async runWithDebugging(
+    _args: ILaunchRequestArguments
   ): Promise<void> {
-    const programText = await this._fileAccessor.readFileAsString(args.program);
-
+    const bps = this.get_breakpoint_ids();
+    this.run(bps);
+  }
+  private async run(bps: number[]): Promise<void> {
     const eventTarget = createDebugConsoleEventTarget();
-    this._compiler
-      .run(programText, args.entry ?? "", args.shots, eventTarget)
-      .then(() => {
+    const res = await this.debugService
+      .evalContinue(bps, eventTarget)
+      .catch((e) => {
+        log.info(`ending session due to error: ${e}`);
         vscode.debug.activeDebugConsole.appendLine("");
         vscode.debug.activeDebugConsole.appendLine(SimulationCompleted);
         this.sendEvent(new TerminatedEvent());
         this.sendEvent(new ExitedEvent(0));
       });
+
```

Comment:
Did you mean to continue on to here if the catch handles the error? e.g. line 214 says it's ending the session, but the catch handler will swallow the error. For example, try the below and you'll notice that both logs happen.

```js
async function foo() {
    await Promise.reject("test").catch(e => console.log("caught the error"));
    console.log("continued after the catch");
}
```